api/helpers/gamestate/deal-cards.js

```diff
@@ -0,0 +1,80 @@
+const GameStatus = require('../../../utils/GameStatus.json');
+const DeckIds = require('../../../utils/DeckIds.json');
+const MoveType = require('../../../utils/MoveType.json');
+const GamePhase = require('../../../utils/GamePhase.json');
+
+module.exports = {
+  friendlyName: 'Deal Cards',
+
+  description: 'Creates initial GameStateRow for specified game, with cards dealt ot both players (5 for p0 and 6 for p1). Errors if Game has previous game or incorrec status.',
+
+  inputs: {
+    game: {
+      type: 'ref',
+      description: 'Game record to be initialized with first GameState',
+      extendedDescription: 'Game is expected to have populated collection of .gameStates',
+      required: true,
+    },
+  },
+
+  fn: async ({ game }, exits) => {
+    
+    try {
+
+      if (game.status !== GameStatus.STARTED) {
+        return exits.error({ message: 'Game has not yet started or is over' });
+      }
+
+      if (!game.gameStates) {
+        return exits.error({ message: 'Game was not populated with .gameStates collection' });
+      }
+
+      if (game.gameStates.length) {
+        return exits.error({ message: 'Cards are already dealt' });
+      }
+
+      const deck = _.shuffle(
+        DeckIds.map((cardId) => sails.helpers.gamestate.convertStrToCard(cardId))
+      );
+
+      const p0 = {
+        hand: deck.splice(0, 5),
+        points: [],
+        faceCards: [],
+      };
+
+      const p1 = {
+        hand: deck.splice(0, 6),
+        points: [],
+        faceCards: [],
+      };
```

Comment:
this is the beauty of the GameState API. Just manipulating arrays in memory before saving and publishing